src/schema/v2/quiz.ts

```diff
@@ -34,6 +35,55 @@ export const QuizType = new GraphQLObjectType<any, ResolverContext>({
         }
       },
     },
+    recommendedArtworks: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ArtworkType))
+      ),
+      resolve: async (
+        { quiz_artworks },
+        _,
+        { savedArtworkLoader, relatedLayerArtworksLoader }
+      ) => {
+        if (!savedArtworkLoader) return []
+
+        try {
+          const artworks = await Promise.all(
+            quiz_artworks.map(({ artwork_id }) => {
+              return savedArtworkLoader(artwork_id)
+            })
+          )
+
+          const savedArtworks = artworks.filter(({ is_saved }) => is_saved)
+
+          if (savedArtworks.length === 0) return []
+
+          const recommendedArtworks = await Promise.all(
```

Comment:
I don't love it bc you are making simultaneous concurrent requests to a very slow  endpoint, it smells of DOS ourselves

Reply:
Also, this isn't unusual. We do this all the time with normal queries.